src/trng/mod.rs

```diff
@@ -1 +1,40 @@
 pub mod registers;
+use crate::memory_map::mmio;
+use registers::Registers;
+
+pub struct TRNG {
+    registers: Registers,
+    pub has_key: bool,
+}
+
+impl TRNG {
+    pub fn init() -> Self {
+        let mut registers = Self {
+            registers: Registers::new(mmio::TRNG),
+            has_key: false,
+        };
+        unsafe {
+            registers.registers.set_generate_key(true);
+        }
+        registers.has_key = true;
+        return registers;
+    }
+
+    pub fn wipe_key(&mut self) {
+        unsafe {
+            self.registers.set_wipe_key(true);
+        }
+    }
+
+    pub fn generate_new_key(&mut self) {
+        unsafe {
+            self.registers.set_generate_key(true);
+        }
+    }
+
+    pub fn get_trng_data(&self) -> u32 {
+        // FIXME use interrupts
```

Comment:
Since the state of the key is could not be generated, maybe an `assert!` or `Result` should be used to ensure key is set properly before asking the hardware for another random number.